teps/0121-refine-retries-for-taskruns-and-customruns.md

```diff
@@ -156,40 +353,174 @@ Three sub-options about the way to implement `retriesStatus`:
+kind: PipelineRun
+metadata:
+  name: ci-pipeline
+spec:
+  pipelineSpec:
+    tasks:
+    - name: loop-task
+      params:
+        - name: test-type
+          value:
+            - codeanalysis
+            - unittests
+            - e2etests
+      taskSpec:
+        apiVersion: custom.tekton.dev/v1alpha1
+        kind: TaskLoop
+        spec:
+          taskRef:
+            name: my-task
+          iterateParam: test-type
+          retries: 2
+```
+
+It would no longer be possible to specify a retries field in the Pipeline Task and have it passed to the
+custom run by reference. For example, with the following PipelineRun: 
+
+```yaml
+apiVersion: custom.tekton.dev/v1alpha1
+kind: TaskLoop
+metadata:
+  name: testloop
+spec:
+  taskRef:
+    name: my-task
+  iterateParam: test-type
+  retries: 2
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: ci-pipeline
+spec:
+  pipelineSpec:
+    tasks:
+    - name: loop-task
+      params:
+        - name: test-type
+          value:
+            - codeanalysis
+            - unittests
+            - e2etests
+      retries: 5
+      taskRef:
+        apiVersion: custom.tekton.dev/v1alpha1
+        kind: TaskLoop
+        name: my-task-loop
+```
+
+The Run that would be created would be:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+spec:
+  params:
+  - name: test-type
+    value:
+    - codeanalysis
+    - unittests
+    - e2etests
+  ref:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: TaskLoop
+    name: testloop
+status:
+  extraFields:
+    taskLoopSpec:
+      iterateParam: test-type
+      retries: 2
+      taskRef:
+        name: my-task
+```
+
+The `retries` specified in the Pipeline Task would not be passed to the custom Run. There would be no ambiguity for the custom run controller, but the user might not know which `retries` field to use.
+
+**Benefits:**
+
+- Consistent interface for `retries`
+- Custom task controller developers get a default implementation of retries for free (by embedding in a pipeline)
+- "Pipelines in pipeline" can be retried the same as the other resources
+- Improve the retries of TaskRuns created from PipelineTasks by using separate TaskRuns for each retry
+- No changes to the PipelineRun or TaskRun spec
+- Cannot accidentally introduce changes that make assumptions about how a CustomRun has implemented RetriesStatus
+
+**Concerns:**
+
+- API Change for `Run` and `CustomRun` (need to remove `retries` & `retriesStatus`)
+  - We are moving Custom Task Run from alpha (Run) to beta (CustomRun) (see [TEP-0114](https://github.com/tektoncd/community/blob/main/teps/0114-custom-tasks-beta.md)), which is a great timing for us to remove fields from `Run`.
+- Dashboard and CLI must be updated to read from PipelineRun status instead of `taskRun.status.retriesStatus` and `customRun.status.retriesStatus`.
+  If a customRun has its own specialized type of retries, this information cannot be reflected on the dashboard.
```

Comment:
Heh, to be honest I think changes to dashboard and CLI will be required in both cases, and unless we enforce the status fields for custom runs, we won't be able to display them in CLI/dashboard with either solution.